scipy/linalg/tests/test_matfuncs.py

```diff
@@ -321,72 +326,76 @@ def test_sqrtm_type_preservation_and_conversion(self):
         for matrix_as_list in (
                 [[1, 0], [0, -1]],
                 [[0, 1], [1, 0]],
-                [[0, 1, 0], [0, 0, 1], [1, 0, 0]]):
+                [[0, 1, 0], [0, 0, -1], [1, 0, 0]]):
 
             # check that the spectrum has the expected properties
             W = scipy.linalg.eigvals(matrix_as_list)
             assert_(any(w.imag or w.real < 0 for w in W))
 
             # check complex->complex
             A = np.array(matrix_as_list, dtype=complex)
-            A_sqrtm, info = sqrtm(A, disp=False)
+            A_sqrtm = sqrtm(A)
             assert_(A_sqrtm.dtype.char in complex_dtype_chars)
 
             # check float->complex
             A = np.array(matrix_as_list, dtype=float)
-            A_sqrtm, info = sqrtm(A, disp=False)
+            A_sqrtm = sqrtm(A)
             assert_(A_sqrtm.dtype.char in complex_dtype_chars)
 
-    def test_blocksizes(self):
-        # Make sure I do not goof up the blocksizes when they do not divide n.
-        np.random.seed(1234)
-        for n in range(1, 8):
-            A = np.random.rand(n, n) + 1j*np.random.randn(n, n)
-            A_sqrtm_default, info = sqrtm(A, disp=False, blocksize=n)
-            assert_allclose(A, np.linalg.matrix_power(A_sqrtm_default, 2))
-            for blocksize in range(1, 10):
-                A_sqrtm_new, info = sqrtm(A, disp=False, blocksize=blocksize)
-                assert_allclose(A_sqrtm_default, A_sqrtm_new)
-
     def test_al_mohy_higham_2012_experiment_1(self):
         # Matrix square root of a tricky upper triangular matrix.
         A = _get_al_mohy_higham_2012_experiment_1()
-        A_sqrtm, info = sqrtm(A, disp=False)
-        A_round_trip = A_sqrtm.dot(A_sqrtm)
+        A_sqrtm = sqrtm(A)
+        A_round_trip = A_sqrtm @ A_sqrtm
         assert_allclose(A_round_trip, A, rtol=1e-5)
         assert_allclose(np.tril(A_round_trip), np.tril(A))
 
     def test_strict_upper_triangular(self):
-        # This matrix has no square root.
+        # This matrix has no square root but upper triangular hence upper
+        # triangle will be filled with junk values.
         for dt in int, float:
             A = np.array([
                 [0, 3, 0, 0],
                 [0, 0, 3, 0],
                 [0, 0, 0, 3],
                 [0, 0, 0, 0]], dtype=dt)
-            A_sqrtm, info = sqrtm(A, disp=False)
-            assert_(np.isnan(A_sqrtm).all())
-
+            with suppress_warnings() as sup:
+                sup.filter(category=LinAlgWarning)
+
+                A_sqrtm = sqrtm(A)
+                assert_allclose(np.tril(A_sqrtm), np.zeros((4, 4)))
+                assert np.isnan(A_sqrtm).any()
+                assert np.isinf(A_sqrtm).any()
+
+    # Future edit: This squareroot is not possible to find algorithmically
+    # with the current methods. Now sqrtm docstring has another example of
+    # such matrix whose squareroot is not a polynomial in it. Hence no need
+    # to test it here.
```

Comment:
Is it possible to actually add a test , if only to fix the behavior on this algorithmic corner case? Whatever the current behavior is

Reply:
The test can be added but the return value will be changing depending on which math library is used (because we are not filling NaNs anymore as mentioned above).

Reply:
I'd then add a test that it does not segfault and returns something.